e2e/lvm/pvc_test.go

```diff
@@ -15,75 +16,199 @@ import (
 				Eventually(func() bool {
-					err := tests.DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: filepvc.Name, Namespace: filepvc.Namespace}, filepvc)
-					return err == nil && filepvc.Status.Phase == k8sv1.ClaimBound
+					err := client.Get(ctx, types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, pvc)
+					return err == nil && pvc.Status.Phase == k8sv1.ClaimBound
 				}, timeout, interval).Should(BeTrue())
-				fmt.Printf("PVC %s is bound\n", filepvc.Name)
+				fmt.Printf("PVC %s is bound\n", pvc.Name)
 
 				Eventually(func() bool {
-					err = tests.DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: filepod.Name, Namespace: filepod.Namespace}, filepod)
-					return err == nil && filepod.Status.Phase == k8sv1.PodRunning
+					err = client.Get(ctx, types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, pod)
+					return err == nil && pod.Status.Phase == k8sv1.PodRunning
 				}, timeout, interval).Should(BeTrue())
-				fmt.Printf("Pod %s is running\n", filepod.Name)
+				fmt.Printf("Pod %s is running\n", pod.Name)
 
-				err = tests.DeployManagerObj.GetCrClient().Delete(context.TODO(), filepod)
+				By("Creating a Snapshot of the block-pvc")
+				snapshot = tests.GetSampleVolumeSnapshot(pvc.Name+"-snapshot", pvc.Name, tests.StorageClass)
+				err = client.Create(ctx, snapshot)
 				Expect(err).To(BeNil())
-				fmt.Printf("Pod %s is deleted\n", filepod.Name)
+				fmt.Printf("Snapshot %s is created\n", snapshot.Name)
 
-				err = tests.DeployManagerObj.GetCrClient().Delete(context.TODO(), filepvc)
+				By("Creating a clone of the block-pvc")
+				clonePvc = tests.GetSamplePvc(size, pvc.Name+"-clone", k8sv1.PersistentVolumeBlock, tests.StorageClass, "PersistentVolumeClaim", pvc.Name)
+				err = client.Create(ctx, clonePvc)
+				Expect(err).To(BeNil())
+				fmt.Printf("Cloned PVC %s is created\n", clonePvc.Name)
+
+				clonePod = tests.GetSamplePod("clone-lvmblockpod", clonePvc.Name)
+				err = client.Create(ctx, clonePod)
 				Expect(err).To(BeNil())
-				fmt.Printf("PVC %s is deleted\n", filepvc.Name)
 
-				By("PVC(block) Should be bound and Pod should be running")
 				Eventually(func() bool {
-					err := tests.DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: blockpvc.Name, Namespace: blockpvc.Namespace}, blockpvc)
-					return err == nil && blockpvc.Status.Phase == k8sv1.ClaimBound
+					err := client.Get(ctx, types.NamespacedName{Name: clonePvc.Name, Namespace: clonePvc.Namespace}, clonePvc)
+					return err == nil && clonePvc.Status.Phase == k8sv1.ClaimBound
 				}, timeout, interval).Should(BeTrue())
-				fmt.Printf("PVC %s is bound\n", blockpvc.Name)
+				fmt.Printf("Cloned PVC %s is bound\n", clonePvc.Name)
+
+				By("Restore Snapshot for block-pvc")
+				restorePvc = tests.GetSamplePvc(size, pvc.Name+"-restore", k8sv1.PersistentVolumeBlock, tests.StorageClass, snapshot.Name, pvc.Name)
```

Comment:
This doesn't look right - shouldn't it be
```suggestion
				restorePvc = tests.GetSamplePvc(size, pvc.Name+"-restore", k8sv1.PersistentVolumeBlock, tests.StorageClass, "VolumeSnapshot", snapshot.Name)
```